Backend performance optimizations

rmeyers4 opened the pull request:
Closes #151

rmeyers4 commented:
@trevorskaggs : Ready for review!

AlexMountain commented on shelter/serializers.py:
```diff
@@ -8,14 +8,14 @@
 
 class RoomSerializer(serializers.ModelSerializer):
-    animals = AnimalSerializer(source='animal_set', many=True, required=False, read_only=True)
+    # animals = AnimalSerializer(source='animal_set', many=True, required=False, read_only=True)
```
We need this one added back in some capacity as well.

rmeyers4 replied:
Nice catch, I got overzealous at the beginning and meant to get to this one. Will restore and address shelter performance in a different ticket.

AlexMountain commented on hotline/serializers.py:
```diff
@@ -78,14 +94,35 @@ class Meta:
+            self.fields.pop('action_history')
+            self.fields.pop('assigned_evac')
+
+        # if fields is not None:
+        #     # Drop any fields that are not specified in the `fields` argument.
+        #     allowed = set(fields)
+        #     existing = set(self.fields)
+        #     for field_name in existing - allowed:
+        #         self.fields.pop(field_name)
```
Are we saving this for later?

rmeyers4 replied:
Ah, this was copy/pasted code from a DRF example, not needed!

AlexMountain reviewed (commented):
Sweet sweet speedy action.  We seem to be missing a couple of returns however causing some pages to explode.  With those added back I think we'll be good!

rmeyers4 commented on animals/serializers.py:
```diff
@@ -5,27 +5,16 @@
     shelter_name = serializers.SerializerMethodField()
     shelter = serializers.SerializerMethodField()
-    is_stray = serializers.SerializerMethodField()
-    evacuation_assignments = serializers.SerializerMethodField()
```
Good call, I added it back to the `AnimalSerializer` and added a prefetch.

rmeyers4 commented:
@AlexMountain : Thanks for the thorough review! I think I addressed the issues you identified. I clicked through the Dispatch process and nothing exploded. I think some additional optimization will be needed for Dispatch and Shelter but those can be addressed in a different ticket.

trevorskaggs reviewed (approved):
SNAP, these are awesome.  Looking forward to seeing in production!

AlexMountain reviewed (approved):
Awesome dude, good to go!
